iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/repair/RepairDataFileScanUtil.java

```diff
@@ -283,10 +282,27 @@ private void checkNonAlignedDeviceSeries(
     if (!checkTsFileResource || actualDeviceStartTime > actualDeviceEndTime) {
       return;
     }
-    TimeRange actualDeviceTimeRange = new TimeRange(actualDeviceStartTime, actualDeviceEndTime);
-    if (!actualDeviceTimeRange.equals(deviceTimeRangeInResource)) {
-      throw new CompactionStatisticsCheckFailedException(
-          device, deviceTimeRangeInResource, actualDeviceTimeRange);
+    compareDeviceTimeRange(
+        device,
+        deviceTimeRangeInResource,
+        new TimeRange(actualDeviceStartTime, actualDeviceEndTime));
+  }
+
+  private void compareDeviceTimeRange(
+      IDeviceID device, TimeRange deviceTimeRangeInResource, TimeRange actualDeviceTimeRange) {
+    long innerCompactionCount = resource.getTsFileID().getInnerCompactionCount();
+    if (innerCompactionCount == 0) {
+      // for the files generate by flush with deletions, the statistics may be larger than the
+      // actual
+      if (!deviceTimeRangeInResource.contains(actualDeviceTimeRange)) {
+        throw new CompactionStatisticsCheckFailedException(
+            device, deviceTimeRangeInResource, actualDeviceTimeRange);
+      }
+    } else {
+      if (!actualDeviceTimeRange.equals(deviceTimeRangeInResource)) {
+        throw new CompactionStatisticsCheckFailedException(
+            device, deviceTimeRangeInResource, actualDeviceTimeRange);
+      }
```

Comment:
The two exceptions seem identical. Is the upper level or the log able to distinguish them (the missing of a device or the mismatch of time range)?